src/System.CommandLine.Benchmarks/CommandLine/Perf_Parser_Simple.cs

```diff
@@ -0,0 +1,35 @@
+﻿using BenchmarkDotNet.Attributes;
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+
+namespace System.CommandLine.Benchmarks.CommandLine
+{
+    [BenchmarkCategory(Categories.CommandLine)]
+    public class Perf_Parser_Simple
+    {
+        [Params(new string[0], new string[4] { "--bool", "true", "-s", "test" })]
+        public string[] Args { get; set; }
+
+        [Benchmark]
+        public int Sync() => BuildCommand().Invoke(Args);
+
+        [Benchmark]
+        public Task<int> Async() => BuildCommand().InvokeAsync(Args);
```

Comment:
A significant chunk of code that could be optimized is inside of the `Invoke` and `InvokeAsync` extension methods on `Command`. These use the default middleware pipeline. This is all opt-in. It represents many of System.CommandLine's features that are not found in many other command line libraries. An apples-to-apples performance comparison to other libraries would omit these until/unless we optimize the middleware pipeline. 

Here's what it's doing:

https://github.com/dotnet/command-line-api/blob/edee3cf6c15e9d3430d8cc4eec10b4928fcf63ce/src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs#L345-L359

Leaving this out (or optimizing it) will knock about 50% off of the allocations. The pipeline is composed together here

https://github.com/dotnet/command-line-api/blob/843fdc7958ec919c70462635b18ac2152a907f67/src/System.CommandLine/Invocation/InvocationPipeline.cs#L42-L68